### PR TITLE
User list changes

### PR DESCRIPTION
**Pull request by Darklol**
feat: endpoints to use in users page

### CODE REVIEW

**Comment by IP-13** [comment]
|Overall Project|60.82% **`-1.22%`**|:green_apple:|
|:-|:-|:-:|
|Files changed|27.78%|:x:|
<br>

|File|Coverage||
|:-|:-|:-:|
|[ProfileController.java](https://github.com/itmo-events-app/event-app/blob/44047a48582bf57c1e1c4dcea468a07016b8151b/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fcontroller%2FProfileController.java)|100%|:green_apple:|
|[EventController.java](https://github.com/itmo-events-app/event-app/blob/44047a48582bf57c1e1c4dcea468a07016b8151b/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fcontroller%2FEventController.java)|100%|:green_apple:|
|[RoleService.java](https://github.com/itmo-events-app/event-app/blob/44047a48582bf57c1e1c4dcea468a07016b8151b/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FRoleService.java)|57.52% **`-6.97%`**|:x:|
|[EventRoleService.java](https://github.com/itmo-events-app/event-app/blob/44047a48582bf57c1e1c4dcea468a07016b8151b/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FEventRoleService.java)|29.43% **`-8.73%`**|:x:|
|[RoleController.java](https://github.com/itmo-events-app/event-app/blob/44047a48582bf57c1e1c4dcea468a07016b8151b/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fcontroller%2FRoleController.java)|9.96% **`-10.39%`**|:x:|

**Comment by IP-13** [comment]
|Overall Project|60.95% **`-1.22%`**|:green_apple:|
|:-|:-|:-:|
|Files changed|27.78%|:x:|
<br>

|File|Coverage||
|:-|:-|:-:|
|[ProfileController.java](https://github.com/itmo-events-app/event-app/blob/99411e832ac07da522bb78cd1eec1666408f42e6/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fcontroller%2FProfileController.java)|100%|:green_apple:|
|[EventController.java](https://github.com/itmo-events-app/event-app/blob/99411e832ac07da522bb78cd1eec1666408f42e6/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fcontroller%2FEventController.java)|100%|:green_apple:|
|[RoleService.java](https://github.com/itmo-events-app/event-app/blob/99411e832ac07da522bb78cd1eec1666408f42e6/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FRoleService.java)|57.52% **`-6.97%`**|:x:|
|[EventRoleService.java](https://github.com/itmo-events-app/event-app/blob/99411e832ac07da522bb78cd1eec1666408f42e6/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FEventRoleService.java)|29.43% **`-8.73%`**|:x:|
|[RoleController.java](https://github.com/itmo-events-app/event-app/blob/99411e832ac07da522bb78cd1eec1666408f42e6/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fcontroller%2FRoleController.java)|9.96% **`-10.39%`**|:x:|

**Comment by IP-13** [comment]
|Overall Project|60.97% **`-1.22%`**|:green_apple:|
|:-|:-|:-:|
|Files changed|27.78%|:x:|
<br>

|File|Coverage||
|:-|:-|:-:|
|[ProfileController.java](https://github.com/itmo-events-app/event-app/blob/f2c14789afb160ee73d65b4b23e1aa8bdd48aa93/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fcontroller%2FProfileController.java)|100%|:green_apple:|
|[EventController.java](https://github.com/itmo-events-app/event-app/blob/f2c14789afb160ee73d65b4b23e1aa8bdd48aa93/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fcontroller%2FEventController.java)|100%|:green_apple:|
|[RoleService.java](https://github.com/itmo-events-app/event-app/blob/f2c14789afb160ee73d65b4b23e1aa8bdd48aa93/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FRoleService.java)|57.52% **`-6.97%`**|:x:|
|[EventRoleService.java](https://github.com/itmo-events-app/event-app/blob/f2c14789afb160ee73d65b4b23e1aa8bdd48aa93/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FEventRoleService.java)|29.43% **`-8.73%`**|:x:|
|[RoleController.java](https://github.com/itmo-events-app/event-app/blob/f2c14789afb160ee73d65b4b23e1aa8bdd48aa93/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fcontroller%2FRoleController.java)|9.96% **`-10.39%`**|:x:|

**Review by rimnvd — APPROVED** [review]

File: src/main/java/org/itmo/eventapp/main/repository/EventRoleRepository.java [hunk 1]
```diff
@@ -27,4 +30,9 @@ public interface EventRoleRepository extends JpaRepository<EventRole, Integer> {
     long deleteByEventId(int eventId);
 
     List<EventRole> findAllByUserIdAndRoleId(Integer userId, Integer roleId);
+
+    @Query("SELECT er FROM EventRole er WHERE er.user.id = :userId and :privilege MEMBER OF er.role.privileges")
```
Comment (IP-13) [inline src/main/java/org/itmo/eventapp/main/repository/EventRoleRepository.java:34]: а нельзя er.user_id сразу? Или в jpql так нельзя

**Review by IP-13 — APPROVED** [review]

**Pull request merged**